README.md

```diff
@@ -1,143 +1,110 @@
-This connector implements CDC features for PostgreSQL by reading WAL events 
-into a buffer that is checked on each call of `Read` after the initial snapshot
-has occurred. If there is a record in the buffer, it grabs and returns that 
-record. If it's empty, it returns `ErrEndData` signaling that Conduit should 
-backoff-retry.
 
- This behavior is enabled by default, but can be turned off by adding 
- `"cdc": "off"` to the Source configuration.
+This connector implements CDC features for PostgreSQL by creating a logical replication slot and a publication that
+listens to changes in the configured table. Every detected change is converted into a record and returned in the call to
+`Read`. If there is no record available at the moment `Read` is called, it blocks until a record is available or the
+connector receives a stop signal.
 
-### CDC  Configuration
-When the connector switches to CDC mode, it attempts to start all necessary 
-connections and runs the initial setup commands to create its logical 
-replication slots and publications. It will connect to an existing slot if one
-with the configured name exists.
+If logical replication isn't available on the PostgreSQL instance, the connector can be configured to use long polling
+by adding `"cdcMode":"longPolling"` to the Source configuration.
 
-The Postgres user specified in the connection URL must have sufficient 
-privileges to run all of these setup commands or it will fail.
+### Logical Replication Configuration
 
-Publication and slot name are user configurable, and must be correctly set. 
-The plugin will do what it can to be smart about publication and slot 
-management, but it can't handle everything.
+When the connector switches to CDC mode, it attempts to run the initial setup commands to create its logical replication
+slot and publication. It will connect to an existing slot if one with the configured name exists.
+
+The Postgres user specified in the connection URL must have sufficient privileges to run all of these setup commands, or
+it will fail.
 
 Example configuration for CDC features:
+
 ```json
 {
- "mode": "cdc",
- "logrepl.publicationName": "meroxademo",
- "logrepl.slotName": "meroxademo",
- "url": "url",
- "key": "key",
- "table": "records",
- "columns": "key,column1,column2,column3"
+  "url": "url",
+  "key": "key",
+  "table": "records",
+  "columns": "key,column1,column2,column3",
+  "cdcMode": "logrepl",
+  "logrepl.publicationName": "meroxademo",
+  "logrepl.slotName": "meroxademo"
 }
 ```
 
-### CDC Event Buffer
-There is a private variable bufferSize that dictates the size of the channel 
-buffer that holds WAL events. If it's full, pushing to that channel will be a 
-blocking operation, and thus execution will stop if the handler for WAL events 
-cannot push into that buffer. That blocking execution could have unknown 
-negative performance consequences, so we should have this be sufficiently high 
-and possibly configured by environment variable.
-
```

Comment:
I find this information useful, especially since it highlights a possible performance bottleneck. It would be good to keep it somewhere.

Reply:
I removed it because this is outdated information, there is no such variable.